app/src/main/java/org/fossasia/openevent/app/qrscan/BarcodeGraphicTracker.java

```diff
@@ -0,0 +1,82 @@
+package org.fossasia.openevent.app.qrscan;
+
+import android.util.SparseArray;
+
+import com.google.android.gms.vision.Detector;
+import com.google.android.gms.vision.Tracker;
+import com.google.android.gms.vision.barcode.Barcode;
+
+import org.fossasia.openevent.app.OrgaApplication;
+import org.fossasia.openevent.app.qrscan.ui.GraphicOverlay;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import io.reactivex.Notification;
+import io.reactivex.subjects.PublishSubject;
+
+/**
+ * Generic tracker which is used for tracking or reading a barcode (and can really be used for
+ * any type of item).  This is used to receive newly detected items, add a graphical representation
+ * to an overlay, update the graphics as the item changes, and remove the graphics when the item
+ * goes away.
+ */
+public class BarcodeGraphicTracker extends Tracker<Barcode> {
+    private GraphicOverlay<BarcodeGraphic> mOverlay;
+    private BarcodeGraphic mGraphic;
+
+    @Inject
+    @Named("barcodeEmitter")
+    PublishSubject<Notification<Barcode>> barcodeEmitter;
+
+    BarcodeGraphicTracker(GraphicOverlay<BarcodeGraphic> overlay, BarcodeGraphic graphic) {
+        mOverlay = overlay;
+        mGraphic = graphic;
+
+        OrgaApplication
+            .getAppComponent(overlay.getContext())
+            .inject(this);
```

Comment:
This should not be needed, add all dependencies in a module and do constructor injection